src/pyodide/python-entrypoint-helper.js

```diff
@@ -201,12 +200,7 @@ try {
       }
     }
   }
-
-  // Store the memory snapshot in the ArtifactBundler so that the validator can read it out.
-  // Needs to happen at the top level because the validator does not perform requests.
-  if (ArtifactBundler.isEwValidating()) {
-    ArtifactBundler.storeMemorySnapshot(getMemoryToUpload());
-  }
+  maybeStoreMemorySnapshot();
```

Comment:
Actually I wonder if it wouldn't be better to keep the comment here. Just to make this function call more visible.